File: Low Level Design (LLD)/DesignPatterns/3. Structural/Observer/Java/Main.java
public class Main {
    public static void main(String[] args) {
        WeatherStation station = new WeatherStation();

        PhoneDisplay phoneDisplay = new PhoneDisplay(station);
        LaptopDisplay laptopDisplay = new LaptopDisplay(station);

        station.setTemperature(25.0f);
        station.setTemperature(30.5f);
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Observer/Cpp/main.cpp
#include "include/WeatherStation.h"
#include "include/PhoneDisplay.h"
#include "include/LaptopDisplay.h"

int main() {
    WeatherStation station;

    PhoneDisplay phoneDisplay(&station);
    LaptopDisplay laptopDisplay(&station);

    station.setTemperature(25.0f);
    station.setTemperature(30.5f);

    return 0;
}
